src/main/java/org/gridsuite/directory/notification/server/DirectoryNotificationWebSocketHandler.java

```diff
@@ -154,18 +152,35 @@ private Flux<WebSocketMessage> heartbeatFlux(WebSocketSession webSocketSession)
                 .pingMessage(dbf -> dbf.wrap((webSocketSession.getId() + "-" + n).getBytes(StandardCharsets.UTF_8))));
     }
 
+    public Flux<WebSocketMessage> receive(WebSocketSession webSocketSession) {
+        return webSocketSession.receive()
+                .doOnNext(webSocketMessage -> {
+                    try {
+                        String wsPayload = webSocketMessage.getPayloadAsText();
+                        //if it's not the heartbeat
+                        if (!(wsPayload.startsWith(webSocketSession.getId() + "-"))) {
+                            LOGGER.debug("Message received : {} by session {}", wsPayload, webSocketSession.getId());
+                            Filters receivedFilters = jacksonObjectMapper.readValue(webSocketMessage.getPayloadAsText(), Filters.class);
+                            //because null is not allowed in ConcurrentHashMap and will cause the websocket to close
+                            if (receivedFilters.getUpdateType() != null) {
+                                webSocketSession.getAttributes().put(FILTER_UPDATE_TYPE, receivedFilters.getUpdateType());
+                            }
+                            if (receivedFilters.getElementUuid() != null) {
+                                webSocketSession.getAttributes().put(FILTER_ELEMENT_UUID, receivedFilters.getElementUuid());
+                            }
+                        }
+                    } catch (JsonProcessingException e) {
+                        LOGGER.error("Unprocessable message received");
+                    }
+                });
+    }
+
     @Override
     public Mono<Void> handle(WebSocketSession webSocketSession) {
-        var uri = webSocketSession.getHandshakeInfo().getUri();
```

Comment:
I changed my mind, let's keep the possibility to put params in the url.

Reply:
Why ? For what case do we need the url param ?